delira/utils/config.py

```diff
@@ -1,13 +1,509 @@
+        item : Any
+            item which should be assigned
+        deepcopy : bool, optional
+            if enabled the item is copied to the config, by default False
+        """
+        config_item = self._to_config(item)
+        if deepcopy:
+            self[key] = copy.deepcopy(config_item)
+        else:
+            self[key] = config_item
+
+    @staticmethod
+    def _create_internal_dict(*args, **kwargs):
+        """
+        Defines how internal dicts should be created. Can be used to easily
+        overwrite subclasses
+
+        Returns
+        -------
+        :class:`Config`
+            new config
+        """
+        return Config(*args, **kwargs)
+
+    @classmethod
+    def _to_config(cls, item):
+        """
+        Convert items to config if they are a dict like object
+        but not already a config
+
+        Parameters
+        ----------
+        item : Any
+            item which is converted
+
+        Returns
+        -------
+        Any
+            return a config is item is dict like, otherwise the item is
+            returned
+        """
+        if isinstance(item, dict) and not isinstance(item, cls):
+            # convert dict to config for additional functionality
+            return cls._create_internal_dict(item)
+        else:
+            return item
+
+    def dump(self, path, formatter=yaml.dump, encoder_cls=Encoder, **kwargs):
+        """
+        Save config to a file and add time stamp to config
+
+        Parameters
+        ----------
+        path : str
+            path where config is saved
+        formatter : callable, optional
+            defines the format how the config is saved, by default yaml.dump
+        encoder_cls : :class:`Encoder`, optional
+            trasforms config to a format which can be formatted by the
+            `formatter`, by default Encoder
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        self._timestamp = now()
+        encoded_self = encoder_cls().encode(self)
+        with open(path, "w") as f:
+            formatter(encoded_self, f, **kwargs)
+
+    def dumps(self, formatter=yaml.dump, encoder_cls=Encoder, **kwargs):
+        """
+        Create a loadable string representation from the config and
+        add time stamp to config
+
+        Parameters
+        ----------
+        formatter : callable, optional
+            defines the format how the config is saved, by default yaml.dump
+        encoder_cls : :class:`Encoder`, optional
+            trasforms config to a format which can be formatted by the
+            `formatter`, by default Encoder
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        self._timestamp = now()
+        encoded_self = encoder_cls().encode(self)
+        return formatter(encoded_self, **kwargs)
+
+    @classmethod
+    def load(cls, path, formatter=yaml.load, decoder_cls=Decoder, **kwargs):
```

Comment:
I am not sure, if I want this to be a class method.

What about making this a usual method and calling self.update on the loaded part and making a separate class method (something like `from_file`), which creates the class and calls this method? I think this would be a bit cleaner?

Reply:
This would imply an extra instruction each time a config needs to be loaded (because the config must be instantiated first and can only be loaded in a second step). 
Do we want this?

Reply:
As long as this isn't a user instruction, this should be fine I guess. This way, we also have the option to update from file